common/src/main/java/org/apache/celeborn/common/network/util/TransportFrameDecoder.java

```diff
@@ -99,7 +99,9 @@ private long decodeFrameSize() {
       bodySize = first.readInt();
       nextFrameSize = msgSize + bodySize;
       totalSize -= HEADER_SIZE;
-      if (!first.isReadable()) {
+      // The msgsize and bodysize of some messages e.g. HEARTBEAT are both 0, so buffers can't be
+      // released
+      if (!first.isReadable() && nextFrameSize != 0) {
```

Comment:
I'm not quite sure whether this will cause netty memory leak. I have struggled a lot to solve TransportFrameDecoder's memory leak in early days.

Reply:
if the heartbeat buf is not released here, it will be released after ```
decodeNext(); if (msg.body() == null) {
        frame.release();
      }
```